File: Data Structures & Algorithms/evaluate-reverse-polish-notation/submission-8.py
class Solution:
    def evalRPN(self, tokens: List[str]) -> int:
        #if we have two stacks one for numebrs one for divsion/sub

        stack = []
        def is_number(s):
            try:
                int(s)
                return True
            except ValueError:
                return False

        for i in tokens:
            if is_number(i):
                stack.append(int(i))
            elif i == '-':
                first = stack[len(stack)-1]
                second = stack[len(stack)-2]
                stack.pop()
                stack.pop()
                stack.append(second-first)   
            elif i == '+':
                first = stack[len(stack)-1]
                second = stack[len(stack)-2]
                stack.pop()
                stack.pop()
                stack.append(first+second)   
            elif i == '*':
                first = stack[len(stack)-1]
                second = stack[len(stack)-2]
                stack.pop()
                stack.pop()
                stack.append(second*first)
            else:
                first = stack[len(stack)-1]
                second = stack[len(stack)-2]
                stack.pop()
                stack.pop()
                stack.append(int(second/first))        

        return int(stack[0])

File: Data Structures & Algorithms/climbing-stairs/submission-8.cpp
class Solution {
public:
    int climbStairs(int n) {
        if (n==1) {
            return 1;
        }
        if (n==2) {
            return 2;
        }
        vector<int> dp(n+1,0);
        dp[n-1] = 1;
        dp[n-2] = 2;

        //[0,0,0,0,0,0,n-2,n-1,n]

        for (int i = n-3;i >= 0;i--) {
            dp[i] = dp[i+1] + dp[i+2];
        }

        return dp[0];
    }
};
